lib/navigation/preprocess_encoders.hpp

```diff
@@ -1,25 +1,42 @@
+#include "consts.hpp"
+
 #include <array>
 #include <cstdint>
 
 #include "core/types.hpp"
+#include "core/logger.hpp"
 
 namespace hyped::navigation {
 class EncodersPreprocessor {
  public:
-  EncodersPreprocessor();
+  //EncodersPreprocessor();
+  EncodersPreprocessor(core::ILogger &logger);
 
-  core::EncoderData processData(const core::EncoderData encoder_data);
+  std::optional<core::EncoderData> processData(const core::EncoderData encoder_data);
 
  private:
-  core::EncoderData detectOutliers(const core::EncoderData encoder_data);
+  std::optional<core::EncoderData> detectOutliers(const core::EncoderData encoder_data);
```

Comment:
Docstrings here pls. It's important that future navigation members know what's going on!